spec/Section 6 -- Execution.md

```diff
@@ -712,6 +720,134 @@ A correct executor must generate the following result for that _selection set_:
 }
 ```
 
+### Field Collection
```

Comment:
I need to reconcile this with the new field collection algorithm, this is currently duplicated due to the merge commit